File: final_dataset_A/generated_code/chatgpt/java/PaymentProcessingEndpoint-chatgpt.java
package com.example.demo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.LinkedHashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/payments")
public class PaymentController {

    @PostMapping("/process")
    public ResponseEntity<Map<String, Object>> processPayment(
            @RequestParam String paymentId,
            @RequestParam double amount) {

        // Simulate payment processing
        Map<String, Object> payment = new LinkedHashMap<>();
        payment.put("paymentId", paymentId);
        payment.put("amount", amount);
        payment.put("status", "SUCCESS");

        return ResponseEntity.ok(payment);
    }
}

File: final_dataset_A/generated_code/chatgpt/python/payment_processing_endpoint-chatgpt.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route("/payments/process", methods=["POST"])
def process_payment():
    payment_id = request.json.get("paymentId")
    amount = request.json.get("amount")

    # Simulate payment processing
    payment = {
        "paymentId": payment_id,
        "amount": amount,
        "status": "SUCCESS"
    }

    return jsonify(payment), 200

if __name__ == "__main__":
    app.run(debug=True)